### PR TITLE
Remove deployment to play store for linux case in deploy_gallery test

### PR DESCRIPTION
**Pull request by keyonghan**
## Description

We are migrating `deploy_gallery` to LUCI, but find the deployment to play store has not been run for a while, and can be removed.
This is to remove deployment to play store for `deploy_gallery` test.

## Related Issues

https://github.com/flutter/flutter/issues/66707

### CODE REVIEW

File: dev/bots/deploy_gallery.sh [hunk 1]
```diff
@@ -35,21 +35,6 @@ if [[ "$OS" == "linux" ]]; then
...
-  if [[ -z "$CIRRUS_PR" && "$CIRRUS_BRANCH" == "dev" && "$version" != *"pre"* ]]; then
-    echo "Deploying Flutter Gallery $version to Play Store..."
-    set +x # Don't echo back the below.
-    if [ -n "$ANDROID_GALLERY_UPLOAD_KEY" ]; then
-      echo "$ANDROID_GALLERY_UPLOAD_KEY" | base64 --decode > /root/.android/debug.keystore
-    fi
-    set -x
-    (
-      cd dev/integration_tests/flutter_gallery/android
-      bundle exec fastlane deploy_play_store
```
Comment (xster) [inline dev/bots/deploy_gallery.sh:48]: you can remove all fastlane related code in the codebase as well. There are some in .cirrus.yaml. You can skip fastlane related setup in the dockerfile as well. Also the various fastlane related files in flutter_gallery/android|ios/fastlane.

Comment (keyonghan) [reply dev/bots/deploy_gallery.sh:48]: Please help take another look.
`Gemfile.lock` was updated after I ran `bundle` after removing `fastlane` codes from `Gemfile`.

File: dev/ci/docker_linux/Gemfile [hunk 2]
```diff
@@ -2,7 +2,3 @@
 # in dev/ci/docker_linux/Dockerfile to retrigger a Docker build on Cirrus.
 
 source "https://rubygems.org"
-
```
Comment (christopherfujino) [inline dev/ci/docker_linux/Gemfile:5]: This PR looks good to me.

As a nit, can you also update the timestamp on the Dockerfile https://github.com/flutter/flutter/blob/master/dev/ci/docker_linux/Dockerfile#L13, to bust the cache and trigger a rebuild of the Dockerfile. I want to make sure that all the linux tests pass without fastlane installed.

**Review by christopherfujino — APPROVED** [review]

File: dev/ci/docker_linux/Gemfile (hunk 2 above)
Comment (keyonghan) [reply dev/ci/docker_linux/Gemfile:5]: Updated.

**Comment by xster** [comment]
LGTM. Thanks!

**Pull request merged**